test/e2e/authorizer/authorizer_test.go

```diff
@@ -0,0 +1,417 @@
+	for {
+		var node yaml.Node
+		err = d.Decode(&node)
+		if errors.Is(err, io.EOF) {
+			break
+		}
+		if err != nil {
+			return fmt.Errorf("error decoding node %w", err)
+		}
+
+		bytes, err := yaml.Marshal(&node)
+		if err != nil {
+			return fmt.Errorf("error marshaling node %w", err)
+		}
+
+		if len(node.Content) == 0 {
+			continue
+		}
+
+		for _, c := range node.Content[0].Content {
+			switch c.Value {
+			case "WorkspaceShard":
+				var v tenancyv1alpha1.WorkspaceShard
+				if err := kubeyaml.Unmarshal(bytes, &v); err != nil {
+					return fmt.Errorf("failed to unmarshal WorkspaceShard: %w", err)
+				}
+				_, err := kcpClient.TenancyV1alpha1().WorkspaceShards().Create(ctx, &v, metav1.CreateOptions{})
+				if err != nil {
+					return fmt.Errorf("failed to create WorkspaceShard: %w", err)
+				}
+			case "Workspace":
+				var v tenancyv1alpha1.Workspace
+				if err := kubeyaml.Unmarshal(bytes, &v); err != nil {
+					return fmt.Errorf("failed to unmarshal Workspace: %w", err)
+				}
+				_, err := kcpClient.TenancyV1alpha1().Workspaces().Create(ctx, &v, metav1.CreateOptions{})
+				if err != nil {
+					return fmt.Errorf("failed to create WorkspaceShard: %w", err)
+				}
+			case "ClusterRole":
+				var v rbacv1.ClusterRole
+				if err := kubeyaml.Unmarshal(bytes, &v); err != nil {
+					return fmt.Errorf("failed to unmarshal ClusterRole: %w", err)
+				}
+				_, err := kubeClient.RbacV1().ClusterRoles().Create(ctx, &v, metav1.CreateOptions{})
+				if err != nil {
+					return fmt.Errorf("failed to create ClusterRole: %w", err)
+				}
+			case "ClusterRoleBinding":
+				var v rbacv1.ClusterRoleBinding
+				if err := kubeyaml.Unmarshal(bytes, &v); err != nil {
+					return fmt.Errorf("failed to unmarshal ClusterRoleBinding: %w", err)
+				}
+				_, err := kubeClient.RbacV1().ClusterRoleBindings().Create(ctx, &v, metav1.CreateOptions{})
+				if err != nil {
+					return fmt.Errorf("failed to create ClusterRoleBinding: %w", err)
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+func TestAuthorizer(t *testing.T) {
+	type clients struct {
+		KubeClient kubernetes.Interface
+		KcpClient  kcp.Interface
+	}
+
+	type testSetup struct {
+		framework.RunningServer
+		kcpExpect framework.RegisterWorkspaceExpectation
+		clients   map[string]*clients
+	}
+
+	usersKCPArgs, err := framework.Users([]framework.User{
+		{
+			Name:   "user-1",
+			UID:    "1111-1111-1111-1111",
+			Token:  "user-1-token",
+			Groups: []string{"team-1"},
+		},
+		{
+			Name:   "user-2",
+			UID:    "1111-1111-1111-1111",
+			Token:  "user-2-token",
+			Groups: []string{"team-2"},
+		},
+		{
+			Name:   "user-3",
+			UID:    "1111-1111-1111-1111",
+			Token:  "user-3-token",
+			Groups: []string{"team-3"},
+		},
+	}).ArgsForKCP(t)
+	if err != nil {
+		t.Error(err)
+		return
+	}
+
+	var testCases = []struct {
+		name string
+		work func(ctx context.Context, t framework.TestingTInterface, server testSetup)
+	}{
+		{
+			name: "create namespace and configmap as user-1",
```

Comment:
This test would appear to be a strict subset of the one that follows. Maybe remove?

Reply:
agreed, let's merge both tests.